### PR TITLE
[BUG] Logging out causes sync error

### PR DESCRIPTION
**Pull request by nambatee**
https://phabricator.wikimedia.org/T191748

Also was causing & fixes https://phabricator.wikimedia.org/T190194 and https://phabricator.wikimedia.org/T189172

### CODE REVIEW

**Comment by coreyfloyd** [comment]
Wow, this is fantastic that you found the cause of all these issues!

File: WMF Framework/ReadingListsSyncOperation.swift [hunk 1]
```diff
@@ -147,7 +147,12 @@ internal class ReadingListsSyncOperation: ReadingListsOperation {
             try moc.save()
         }
         
-
+        guard readingListsController.isLoggedIn else {
```
Comment (coreyfloyd) [inline WMF Framework/ReadingListsSyncOperation.swift]: 💥

File: Wikipedia/Code/WMFAppViewController.m [hunk 2]
```diff
@@ -784,6 +784,8 @@ - (void)pauseApp {
     self.settingsViewController = nil;
 
     [self.dataStore.feedContentController stopContentSources];
+    
+    [[WMFAuthenticationManager sharedInstance] deleteLoginTokensAndBrowserCookies];
```
Comment (coreyfloyd) [inline Wikipedia/Code/WMFAppViewController.m]: Why do this?

Just curious what the effect is

Comment (coreyfloyd) [reply Wikipedia/Code/WMFAppViewController.m]: nm… you removed it entirely

File: Wikipedia/Code/WMFAuthenticationManager.swift [hunk 3]
```diff
@@ -177,21 +177,9 @@ class WMFAuthenticationManager: NSObject {
      *  Logs out any authenticated user and clears out any associated cookies
      */
     @objc public func logout(completion: @escaping () -> Void = {}){
-        self.resetLocalUserLoginSettings()
-        completion()
-    }
-    
-    @objc public func deleteLoginTokensAndBrowserCookies() {
```
Comment (coreyfloyd) [inline Wikipedia/Code/WMFAuthenticationManager.swift]: Why is it ok to kill this method? Does it not need to happen or does it happen in another way now?

Comment (nambatee) [reply Wikipedia/Code/WMFAuthenticationManager.swift]: I moved it back where it was, to `logout`, so there's no need for a separate method anymore

**Comment by coreyfloyd** [comment]
Looks good!

**Pull request merged**